### ISSUE TITLE
False positive "Unused LANGUAGE pragma"

### ISSUE THREAD

**Issue by cblp** [ndmitchell/hlint#1650]
```haskell
{-# LANGUAGE QuasiQuotes #-}
[qq|
  ...
|]
```
gives
```
example.hs:1:1-28: Warning: Unused LANGUAGE pragma
Found:
  {-# LANGUAGE QuasiQuotes #-}
Perhaps you should remove it.
```
But without the pragma
```hs
[qq|
  ...
|]
```
```
TypicalUsage.hs:2:3-5: Error: Parse error: on input `...'
Found:
    [qq|
  >   ...
      |]
```

----

```
$ hlint --version
HLint v3.10, (C) Neil Mitchell 2006-2025
```

**Comment by cblp** [comment]
With master version, the same, and also
```
example.hs:1:1-28: Warning: Avoid restricted extensions
Found:
  {-# LANGUAGE QuasiQuotes #-}
Note: may break the code
```

Why is it restricted? What does it mean? I cant find anything about it in GHC docs.

**Comment by cblp** [comment]
okay, found
```hs
badExtensions =
  ...
  , QuasiQuotes -- breaks [x| ...], making whitespace free list comps break
```

Okay, but Hlint breaks `[x| ... |]`, and I always put a space when I type a list comprehension. So what do you recommend instead of QQ to get the result?
